File: invert.go
package main

import (
	"fmt"
	"strconv"
)

func invertList(_list []int) []int {
	length := len(_list)
	counter := length/2

	for i := 0; i < int(counter); i++ {
		fmt.Println(i)
		toCut := i+1
		number, _ := strconv.ParseInt(fmt.Sprintf("%d", length-toCut), 10, 64)
		fmt.Printf(" \n %v \n", number)

		_list[i], _list[number] = _list[number], _list[i]
		
	}

	return _list
}

func main() {
	fmt.Println(invertList([]int{2, 4, 1, 1, 3, 5, 21, 45}))
}

File: invert.py
# A function that inverts a list

# Input: [1, 2, 4, 6, 12, 90, 104, 56, 72]
# Output: [72, 56, 104, 90, 12, 6, 4, 2, 1]

def invert_list(_list):
	length = len(_list)

	"""	if length % 2 == 0:
		is_odd = False
	else:
		is_odd  = True"""
	counter = length/2

	print(counter)

	for i in range(int(counter)):
		print(i)
		number = int(f"-{i + 1}")
		_list[i], _list[number] = _list[number]  ,_list[i]
	return _list

print(invert_list([1, 2, 4, 6, 12, 90, 104, 56, 72]))